File: 2.Dagger/Subcomponents/AnnotationParameter/src/main/java/com/app/annotationparameter/fragment/FragmentModule.java
package com.app.annotationparameter.fragment;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
public class FragmentModule {
    @Provides
    @Named("fragment string")
    String provideStringFromFragment(){
        return "fragment string";
    }
}

File: 2.Dagger/Hilt/app/src/main/java/com/app/hilt/fragment/FragmentModule.kt
package com.app.hilt.fragment

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.FragmentComponent

@Module
@InstallIn(FragmentComponent::class)
class FragmentModule {
    @FragmentString
    @Provides
    fun provideString():String{
        return "String from fragment module"
    }
}